### ISSUE TITLE
Error when encountering variables in Docker Buildx mounts

### ISSUE THREAD

**Issue by ivy** [hadolint/hadolint#705]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

Linting a `Dockerfile` that includes variables in mounts shouldn't result in a syntax error.

### Actual behavior

Hadolint exits with the following error (see example `Dockerfile` below):

```
Dockerfile:3:50 unexpected '$' expecting positive number
```

### Steps to reproduce the behavior

**Output of `hadolint --version`:** `Haskell Dockerfile Linter 2.6.1-no-git`

```bash
# Built with:
docker buildx build - < Dockerfile
# Linted with:
hadolint Dockerfile
```

**Dockerfile** (if relevant)

```Dockerfile
FROM alpine:latest
ENV UID=0 GID=0
RUN --mount=type=cache,target=/var/cache/foo,uid=$UID,gid=$GID
```

**Comment by m-ildefons** [comment]
Hi Ivy,
this is indeed a bug in the parser. I have made a fix, but may take a couple of days before it is merged because it is a breaking change.
Thanks for reporting the bug though.

Cheers

**Comment by ivy** [comment]
Yay! Thank you for working on a fix. 🙌
